src/memkind_bandwidth.c

```diff
@@ -257,20 +261,22 @@ int bandwidth_set_closest_numanode(int num_unique,
             min_unique = 1;
             for (j = 0; j < match.num_numanodes; ++j) {
                 old_errno = errno;
-                distance = numa_distance(numa_node_of_cpu(i),
-                                         match.numanodes[j]);
+                distance = numa_distance(numa_node_of_cpu(i), match.numanodes[j]);
                 errno = old_errno;
                 if (distance < min_distance) {
                     min_distance = distance;
-                    closest_numanode[i] = match.numanodes[j];
+                    if (min_unique > 1) {
```

Comment:
This took me a while to understand.

So, I wouldn't say this is a matrix, but rather an array of closest numa nodes for each CPU. And your arrays don't have the same length, so it doesn't even fit the mathematical definition of an array. And it's not even a CPU matrix like from numactl, so it's doubly confusing.
I'd probably do something like:

struct cpu_memory_nodes {
  Vec<node> nodes;
};
cpu.nodes.sort(|a, b| a.bandwidth.compare(b.bandwidth)).filter(|n| n.bandwidth > lowest_found);

(pseudo-functional code)